### PR TITLE
Breaking Change: Be compatible with Prettier CLI

### PR DESCRIPTION
**Pull request by tunnckoCore**
**What**: Instead of always thinking what was the right flag we can just be compatible with Prettier CLI. So,
instead of 

```
prettier-eslint --prettier.semi=false --prettier.singleQuote
```

we can just do 

```
prettier-eslint --no-semi --single-quote
```
which is 1:1 with `prettier --no-semi --single-quote`

**Why**: It started at https://github.com/jonlabelle/SublimeJsPrettier/issues/35 today. I was thinking that about few months or so. The whole thing is that we can just be a mirror of Prettier's CLI, so in such scenarios like SublimeJsPrettier we can just provide option there (like in prettier-vscode eslintIntegration true/false) and instead of the hardcoded `prettier` in few places, it can use `prettier-eslint`.
Also, it won't be problem to add one more entry in the `bin` field here.

***

As side note want to say, that that Contributing guide and scripts/workflow is kinda good and i like it :) My is almost the same but is a bit more restrictive and locked (not good for newcommers ;d) 

Cheers :beer:

### CODE REVIEW

File: src/format-files.test.js [hunk 1]
```diff
@@ -116,7 +116,11 @@ test('logs errors to the console if something goes wrong', async () => {
   const errorPrefix = expect.stringMatching(/prettier-eslint-cli.*ERROR/)
   const cliError = expect.stringContaining('eslint-config-error')
   const errorOutput = expect.stringContaining('Some weird eslint config error')
-  expect(console.error).toHaveBeenCalledWith(errorPrefix, cliError, errorOutput)
+  expect(console.error).toHaveBeenCalledWith(
```
Comment (tunnckoCore) [inline src/format-files.test.js]: Wat?! Isn't that strange? :D

File: src/parser.js [hunk 2]
```diff
@@ -4,14 +4,16 @@ import findUp from 'find-up'
...
 const logger = getLogger({prefix: 'prettier-eslint-cli'})
 
 const parser = yargs
-  .usage('Usage: $0 <globs>... [--option-1 option-1-value --option-2]')
+  .usage(
+    `Usage: $0 <globs>... [--option-1 option-1-value --option-2]
+
+Prefix an option with "no-" to set it to false, such as --no-semi to disable
+semicolons and --no-eslint-ignore to disable default ignores.`,
+  )
```
Comment (kentcdodds) [inline src/parser.js]: I find deindented multi-line strings distracting. Could we use `stripIndent` from [common tags](https://www.npmjs.com/package/common-tags) (already imported)?

File: src/parser.js [hunk 3]
```diff
@@ -68,13 +76,70 @@ const parser = yargs
...
+    },
+    'tab-width': {
+      default: 2,
+      type: 'number',
+      describe: 'Specify the number of spaces per indentation-level.',
+    },
+    'trailing-comma': {
+      default: 'none',
+      type: 'string',
+      describe: `Print trailing commas wherever possible.
```
Comment (kentcdodds) [inline src/parser.js]: Same here. So it'd be:

```javascript
      describe: stripIndent`
        Print trailing commas wherever possible.

        Valid options:
          - "none" - no trailing commas
          - "es5" - trailing commas where valid in ES5 (objects, arrays, etc)
          - "all" - trailing commas wherever possible (function arguments)
      `,
```

File: src/format-files.js [hunk 4]
```diff
@@ -37,15 +37,29 @@ function formatFilesFromArgv({
   ignore: ignoreGlobs = [],
   eslintIgnore: applyEslintIgnore = true,
   prettierLast,
-  prettier,
+  ...prettierOptions
 }) {
+  /*
```
Comment (kentcdodds) [inline src/format-files.js]: Let's go ahead and just remove this comment. Thanks!

File: cli-test/tests/index.js [hunk 5]
```diff
@@ -27,10 +27,13 @@ test('help outputs usage information and flags', async () => {
   expect(stdout).toContain('Options:\n')
   // just a sanity check.
   // If it's ever longer than 2000 then we've probably got a problem...
-  if (stdout.length > 2000) {
+  if (stdout.length > 4100) {
     console.error(stdout)
     throw new Error(
-      'We probably have a problem. The --help output is probably too long...',
+      `
```
Comment (kentcdodds) [inline cli-test/tests/index.js]: Could you use `oneLine` from `common-tags`?

```
throw new Error(
  oneLine`
    We probably have a problem.
    The --help output is probably too long (${stdout.length})...
  `
)
```

File: src/format-files.test.js (hunk 1 above)
Comment (kentcdodds) [inline src/format-files.test.js]: The prettier-eslint git hook. My guess is prettier was updated since the last time this code was formatted. No worries. I don't mind unrelated formatting changes because prettier's doing it and I <3 prettier :)

Comment (tunnckoCore) [inline src/format-files.test.js]: yea, originally it was the same way, then after the first commits it transformed it to oneline, then back again..

okey.

File: src/parser.js (hunk 2 above)
Comment (tunnckoCore) [inline src/parser.js]: same here :+1: i just forgot, it was in my mind

**Comment by tunnckoCore** [comment]
Okey, i'll fix that things in a bit!

> This is looking super awesome!

Thanks. I'm glad I contributed to some of your packages, because seen why some ESLint rules are very important (that are not included in StandardJS) and leads to better code.

**Comment by tunnckoCore** [comment]
Cool. Everything is okey now :)

Btw, i'm using the PR in the Vim new ALE plugin feature called "ALEFix" which is  asynchronous formatting and is fantastic! For example, i'm new to vim and when tried Neoformat i was shocked that it is blocking ;d

So yea, it works :) Then we can contrib to SublimeJsPrettier to add option. Otherwise i'll just hard code it.

**Comment by kentcdodds** [comment]
Shoot, I made a mistake merging this. Standby...

**Comment by kentcdodds** [comment]
Ok, so, I have everything automated for releases, but it's based on the commit message and when I merged this I messed up the commit message. It's been resolved and there will be a new version published momentarily :) Thanks!

**Comment by tunnckoCore** [comment]
Haha, yea. That's why i don't like `semantic-release`. It is too automated. :D

Thanks again.

**Comment by kentcdodds** [comment]
Without it, this release wouldn't have happened until tomorrow. Being able to release stuff from my phone is amazing. Totally worth a bump every now and then...

**Comment by tunnckoCore** [comment]
> Being able to release stuff from my phone is amazing

Haha, that's crazy dude :D
